generative/maisi/scripts/sample.py

```diff
@@ -0,0 +1,953 @@
+    if (output_size[0] not in [256, 384, 512]) or (output_size[2] not in [128, 256, 384, 512, 640, 768]):
+        raise ValueError(
+            f"The output_size[0] have to be chosen from [256, 384, 512], and output_size[2] have to be chosen from [128, 256, 384, 512, 640, 768], yet got {output_size}."
+        )
+
+    if spacing[0] != spacing[1]:
+        raise ValueError(f"The first two components of spacing need to be equal, yet got {spacing}.")
+    if spacing[0] < 0.5 or spacing[0] > 3.0 or spacing[2] < 0.5 or spacing[2] > 5.0:
+        raise ValueError(
+            f"spacing[0] have to be between 0.5 and 3.0 mm, spacing[2] have to be between 0.5 and 5.0 mm, yet got {spacing}."
+        )
+
+    # check controllable_anatomy_size format
+    if len(controllable_anatomy_size) > 10:
+        raise ValueError(
+            f"The length of list controllable_anatomy_size has to be less than 10. Yet got length equal to {len(controllable_anatomy_size)}."
+        )
+    available_controllable_organ = [
+        "liver",
+        "gallbladder",
+        "stomach",
+        "pancreas",
+        "colon",
+    ]
+    available_controllable_tumor = [
+        "hepatic tumor",
+        "bone lesion",
+        "lung tumor",
+        "colon cancer primaries",
+        "pancreatic tumor",
+    ]
+    available_controllable_anatomy = available_controllable_organ + available_controllable_tumor
+    controllable_tumor = []
+    controllable_organ = []
+    for controllable_anatomy_size_pair in controllable_anatomy_size:
+        if controllable_anatomy_size_pair[0] not in available_controllable_anatomy:
+            raise ValueError(
+                f"The controllable_anatomy have to be chosen from {available_controllable_anatomy}, yet got {controllable_anatomy_size_pair[0]}."
+            )
+        if controllable_anatomy_size_pair[0] in available_controllable_tumor:
+            controllable_tumor += [controllable_anatomy_size_pair[0]]
+        if controllable_anatomy_size_pair[0] in available_controllable_organ:
+            controllable_organ += [controllable_anatomy_size_pair[0]]
+        if controllable_anatomy_size_pair[1] == -1:
+            continue
+        if controllable_anatomy_size_pair[1] < 0 or controllable_anatomy_size_pair[1] > 1.0:
+            raise ValueError(
+                f"The controllable size scale have to be between 0 and 1,0, or equal to -1, yet got {controllable_anatomy_size_pair[1]}."
+            )
+    if len(controllable_tumor + controllable_organ) != len(list(set(controllable_tumor + controllable_organ))):
+        raise ValueError(f"Please do not repeat controllable_anatomy. Got {controllable_tumor + controllable_organ}.")
+    if len(controllable_tumor) > 1:
+        raise ValueError(f"Only one controllable tumor is supported. Yet got {controllable_tumor}.")
+
+    if len(controllable_anatomy_size) > 0:
+        print(
```

Comment:
It would be more preferable to use the `logging` module over print so that we can tune the level of log in notebooks/tutorials. Thanks!

Reply:
Done, thanks